modules/asg/main.tf

```diff
@@ -0,0 +1,81 @@
+data "aws_ami" "amazon-linux-2" {
+  most_recent = true
+  owners      = ["amazon"]
+
+  filter {
+    name   = "name"
+    values = ["amzn2-ami-hvm-*-x86_64-ebs"]
+  }
+}
+
+resource "aws_launch_template" "t3micro-template" {
+  name_prefix   = "t3micro"
+  image_id      = "ami-0359b3157f016ae46"
```

Comment:
Please use aws_ami datasource instead of hardcoding image_id:
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/ami